pkg/controller/template/render.go

```diff
@@ -294,7 +294,9 @@ func generateMachineConfigForName(config *RenderConfig, role, name, templateDir,
 		return nil, fmt.Errorf("error creating MachineConfig from Ignition config: %w", err)
 	}
 	// And inject the osimageurl here
-	mcfg.Spec.OSImageURL = config.OSImageURL
+	// TODO(jkyros): if we're allowing overrides, this makes it hard to figure out of the user overrode it, or if it's just
+	// "passthrough" from here since we don't mark "system" vs "user" machineconfigs.
```

Comment:
Yeah.  Well...presumably we'll revisit this when we add a CRD or something else.

That said really short term we could introduce `hotfixOverrideOSImage` into machineconfig or so...did that ever get debated versus allowing htem to set `osImageURL`?